tests/__init__.py

```diff
@@ -213,11 +214,21 @@ def nocompat_test_suite():
     return result
 
 
+@patch("dulwich.protocol.DEFAULT_GIT_PROTOCOL_VERSION_FETCH", new=0)
+def compat_test_suite_git_protocol_v0():
```

Comment:
Is this running all compat tests twice, not just the client ones?

This should really just be specific to the client code.